File: base/--build/tmp/kapt3/stubs/debug/com/shopify/shopifyapp/cartsection/adapters/CartListAdapter.java
package com.baytrendy.shopifyapp.cartsection.adapters;

import java.lang.System;

@kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000`\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\u0018\u0002\n\u0002\b\u0002\n\u0002\u0010!\n\u0002\u0018\u0002\n\u0002\b\u0005\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010\b\n\u0000\n\u0002\u0010\u0006\n\u0002\b\u0003\n\u0002\u0010\t\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0002\b\u0003\n\u0002\u0018\u0002\n\u0000\n\u0002\u0010 \n\u0002\u0018\u0002\n\u0002\b\u0002\u0018\u00002\b\u0012\u0004\u0012\u00020\u00020\u0001:\u0001%B\u0007\b\u0007\u00a2\u0006\u0002\u0010\u0003J\u0016\u0010\u000f\u001a\u00020\u00102\u0006\u0010\u0011\u001a\u00020\u00122\u0006\u0010\u0013\u001a\u00020\u0012J\b\u0010\u0014\u001a\u00020\u0010H\u0016J\u0010\u0010\u0015\u001a\u00020\u00162\u0006\u0010\u0017\u001a\u00020\u0010H\u0016J\u0018\u0010\u0018\u001a\u00020\u00192\u0006\u0010\u001a\u001a\u00020\u00022\u0006\u0010\u0017\u001a\u00020\u0010H\u0016J\u0018\u0010\u001b\u001a\u00020\u00022\u0006\u0010\u001c\u001a\u00020\u001d2\u0006\u0010\u001e\u001a\u00020\u0010H\u0016J\u001e\u0010\t\u001a\u00020\u00192\f\u0010\u0004\u001a\b\u0012\u0004\u0012\u00020\u00060\u00052\b\u0010\r\u001a\u0004\u0018\u00010\u000eJ&\u0010\u001f\u001a\u00020\u00192\u0006\u0010 \u001a\u00020!2\u0006\u0010\u001a\u001a\u00020\u00022\f\u0010\"\u001a\b\u0012\u0004\u0012\u00020$0#H\u0002R\"\u0010\u0004\u001a\n\u0012\u0004\u0012\u00020\u0006\u0018\u00010\u0005X\u0086\u000e\u00a2\u0006\u000e\n\u0000\u001a\u0004\b\u0007\u0010\b\"\u0004\b\t\u0010\nR\u0010\u0010\u000b\u001a\u0004\u0018\u00010\fX\u0082\u000e\u00a2\u0006\u0002\n\u0000R\u0010\u0010\r\u001a\u0004\u0018\u00010\u000eX\u0082\u000e\u00a2\u0006\u0002\n\u0000\u00a8\u0006&"}, d2 = {"Lcom/shopify/shopifyapp/cartsection/adapters/CartListAdapter;", "Landroidx/recyclerview/widget/RecyclerView$Adapter;", "Lcom/shopify/shopifyapp/cartsection/viewholders/CartItem;", "()V", "data", "", "Lcom/shopify/buy3/Storefront$CheckoutLineItemEdge;", "getData", "()Ljava/util/List;", "setData", "(Ljava/util/List;)V", "layoutInflater", "Landroid/view/LayoutInflater;", "model", "Lcom/shopify/shopifyapp/cartsection/viewmodels/CartListViewModel;", "getDiscount", "", "regular", "", "special", "getItemCount", "getItemId", "", "position", "onBindViewHolder", "", "holder", "onCreateViewHolder", "parent", "Landroid/view/ViewGroup;", "viewType", "setVariants", "item", "Lcom/shopify/shopifyapp/cartsection/models/CartListItem;", "selectedOptions", "", "Lcom/shopify/buy3/Storefront$SelectedOption;", "ClickHandlers", "base_debug"})
public final class CartListAdapter extends androidx.recyclerview.widget.RecyclerView.Adapter<com.baytrendy.shopifyapp.cartsection.viewholders.CartItem> {
    @org.jetbrains.annotations.Nullable()
    private java.util.List<com.shopify.buy3.Storefront.CheckoutLineItemEdge> data;
    private android.view.LayoutInflater layoutInflater;
    private com.baytrendy.shopifyapp.cartsection.viewmodels.CartListViewModel model;
    
    @org.jetbrains.annotations.Nullable()
    public final java.util.List<com.shopify.buy3.Storefront.CheckoutLineItemEdge> getData() {
        return null;
    }
    
    public final void setData(@org.jetbrains.annotations.Nullable()
    java.util.List<com.shopify.buy3.Storefront.CheckoutLineItemEdge> p0) {
    }
    
    @org.jetbrains.annotations.NotNull()
    @java.lang.Override()
    public com.baytrendy.shopifyapp.cartsection.viewholders.CartItem onCreateViewHolder(@org.jetbrains.annotations.NotNull()
    android.view.ViewGroup parent, int viewType) {
        return null;
    }
    
    @java.lang.Override()
    public long getItemId(int position) {
        return 0L;
    }
    
    @java.lang.Override()
    public void onBindViewHolder(@org.jetbrains.annotations.NotNull()
    com.baytrendy.shopifyapp.cartsection.viewholders.CartItem holder, int position) {
    }
    
    private final void setVariants(com.baytrendy.shopifyapp.cartsection.models.CartListItem item, com.baytrendy.shopifyapp.cartsection.viewholders.CartItem holder, java.util.List<? extends com.shopify.buy3.Storefront.SelectedOption> selectedOptions) {
    }
    
    @java.lang.Override()
    public int getItemCount() {
        return 0;
    }
    
    public final void setData(@org.jetbrains.annotations.NotNull()
    java.util.List<com.shopify.buy3.Storefront.CheckoutLineItemEdge> data, @org.jetbrains.annotations.Nullable()
    com.baytrendy.shopifyapp.cartsection.viewmodels.CartListViewModel model) {
    }
    
    public final int getDiscount(double regular, double special) {
        return 0;
    }
    
    @javax.inject.Inject()
    public CartListAdapter() {
        super();
    }
    
    @kotlin.Metadata(mv = {1, 1, 16}, bv = {1, 0, 3}, k = 1, d1 = {"\u0000 \n\u0002\u0018\u0002\n\u0002\u0010\u0000\n\u0002\b\u0002\n\u0002\u0010\u0002\n\u0000\n\u0002\u0018\u0002\n\u0000\n\u0002\u0018\u0002\n\u0002\b\u0004\b\u0086\u0004\u0018\u00002\u00020\u0001B\u0005\u00a2\u0006\u0002\u0010\u0002J\u0016\u0010\u0003\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\bJ\u0016\u0010\t\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\bJ\u0016\u0010\n\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\bJ\u0016\u0010\u000b\u001a\u00020\u00042\u0006\u0010\u0005\u001a\u00020\u00062\u0006\u0010\u0007\u001a\u00020\b\u00a8\u0006\f"}, d2 = {"Lcom/shopify/shopifyapp/cartsection/adapters/CartListAdapter$ClickHandlers;", "", "(Lcom/shopify/shopifyapp/cartsection/adapters/CartListAdapter;)V", "decrease", "", "view", "Landroid/view/View;", "item", "Lcom/shopify/shopifyapp/cartsection/models/CartListItem;", "increase", "moveToWishList", "removeFromCart", "base_debug"})
    public final class ClickHandlers {
        
        public final void moveToWishList(@org.jetbrains.annotations.NotNull()
        android.view.View view, @org.jetbrains.annotations.NotNull()
        com.baytrendy.shopifyapp.cartsection.models.CartListItem item) {
        }
        
        public final void removeFromCart(@org.jetbrains.annotations.NotNull()
        android.view.View view, @org.jetbrains.annotations.NotNull()
        com.baytrendy.shopifyapp.cartsection.models.CartListItem item) {
        }
        
        public final void increase(@org.jetbrains.annotations.NotNull()
        android.view.View view, @org.jetbrains.annotations.NotNull()
        com.baytrendy.shopifyapp.cartsection.models.CartListItem item) {
        }
        
        public final void decrease(@org.jetbrains.annotations.NotNull()
        android.view.View view, @org.jetbrains.annotations.NotNull()
        com.baytrendy.shopifyapp.cartsection.models.CartListItem item) {
        }
        
        public ClickHandlers() {
            super();
        }
    }
}

File: base/src/main/java/com/baytrendy/shopifyapp/cartsection/adapters/CartListAdapter.kt
package com.baytrendy.shopifyapp.cartsection.adapters

import android.graphics.Paint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView


import com.shopify.buy3.Storefront
import com.baytrendy.shopifyapp.R
import com.baytrendy.shopifyapp.databinding.MCartitemBinding
import com.baytrendy.shopifyapp.basesection.models.CommanModel
import com.baytrendy.shopifyapp.cartsection.models.CartListItem
import com.baytrendy.shopifyapp.cartsection.viewholders.CartItem
import com.baytrendy.shopifyapp.cartsection.viewmodels.CartListViewModel
import com.baytrendy.shopifyapp.utils.CurrencyFormatter

import java.math.BigDecimal

import javax.inject.Inject

class CartListAdapter @Inject constructor() : RecyclerView.Adapter<CartItem>() {
    var data: MutableList<Storefront.CheckoutLineItemEdge>? = null
    private var layoutInflater: LayoutInflater? = null
    private var model: CartListViewModel? = null

    init {
        setHasStableIds(true)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CartItem {
        val binding = DataBindingUtil.inflate<MCartitemBinding>(LayoutInflater.from(parent.context), R.layout.m_cartitem, parent, false)
        return CartItem(binding)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun onBindViewHolder(holder: CartItem, position: Int) {
        val item = CartListItem()
        item.position = position
        item.product_id = data?.get(position)!!.node.variant.product.id.toString()
        item.variant_id = data?.get(position)!!.node.variant.id.toString()
        item.productname = data?.get(position)!!.node.title
        val variant = data?.get(position)!!.node.variant
        item.normalprice = CurrencyFormatter.setsymbol(variant.presentmentPrices.edges[0].node.price.amount, variant.presentmentPrices.edges[0].node.price.currencyCode.toString())
        if (variant.compareAtPriceV2 != null) {
            val special = java.lang.Double.valueOf(variant.presentmentPrices.edges[0].node.compareAtPrice.amount)
            val regular = java.lang.Double.valueOf(variant.presentmentPrices.edges[0].node.price.amount)
            if (BigDecimal.valueOf(special).compareTo(BigDecimal.valueOf(regular)) == 1) {
                item.normalprice = CurrencyFormatter.setsymbol(variant.presentmentPrices.edges[0].node.compareAtPrice.amount, variant.presentmentPrices.edges[0].node.compareAtPrice.currencyCode.toString())
                item.specialprice = CurrencyFormatter.setsymbol(variant.presentmentPrices.edges[0].node.price.amount, variant.presentmentPrices.edges[0].node.price.currencyCode.toString())
                item.offertext = getDiscount(special, regular).toString() + "%off"

            } else {
                item.normalprice = CurrencyFormatter.setsymbol(variant.presentmentPrices.edges[0].node.price.amount, variant.presentmentPrices.edges[0].node.price.currencyCode.toString())
                item.specialprice = CurrencyFormatter.setsymbol(variant.presentmentPrices.edges[0].node.compareAtPrice.amount, variant.presentmentPrices.edges[0].node.compareAtPrice.currencyCode.toString())
                item.offertext = getDiscount(regular, special).toString() + "%off"
            }
            holder.binding.regularprice.paintFlags = holder.binding.regularprice.paintFlags or Paint.STRIKE_THRU_TEXT_FLAG
            holder.binding.specialprice.visibility = View.VISIBLE
            holder.binding.offertext.visibility = View.VISIBLE
            holder.binding.offertext.setTextColor(holder.binding.offertext.context.resources.getColor(R.color.green))
        } else {
            holder.binding.specialprice.visibility = View.GONE
            holder.binding.offertext.visibility = View.GONE
            holder.binding.regularprice.paintFlags = holder.binding.regularprice.paintFlags and Paint.STRIKE_THRU_TEXT_FLAG.inv()
        }
        val model = CommanModel()
        model.imageurl = variant?.image?.originalSrc
        holder.binding.commondata = model
        item.image = variant?.image?.originalSrc
        item.qty = data?.get(position)!!.node.quantity!!.toString()
        holder.binding.name.textSize = 12f
        holder.binding.specialprice.textSize = 13f
        holder.binding.regularprice.textSize = 11f
        holder.binding.offertext.textSize = 11f
        holder.binding.variantOne.textSize = 11f
        holder.binding.variantTwo.textSize = 11f
        holder.binding.variantThree.textSize = 11f
        holder.binding.remove.textSize = 11f
        holder.binding.movetowish.textSize = 11f
        holder.binding.quantity.textSize = 11f
        holder.binding.handlers = ClickHandlers()
        setVariants(item, holder, variant.selectedOptions)

    }

    private fun setVariants(item: CartListItem, holder: CartItem, selectedOptions: List<Storefront.SelectedOption>) {
        try {
            val iterator1 = selectedOptions.iterator()
            var counter = 0
            var option: Storefront.SelectedOption
            while (iterator1.hasNext()) {
                counter = counter + 1
                option = iterator1.next()
                if (!option.value.equals("Default Title", true)) {
                    val finalvalue = option.name + " : " + option.value
                    if (counter == 1) {
                        item.variant_one = finalvalue
                    }
                    if (counter == 2) {
                        item.variant_two = finalvalue
                    }
                    if (counter == 3) {
                        item.variant_three = finalvalue
                    }
                    if (counter > 3) {
                        break
                    }
                }
            }
            holder.binding.variantdata = item
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun getItemCount(): Int {
        return data!!.size
    }

    fun setData(data: MutableList<Storefront.CheckoutLineItemEdge>, model: CartListViewModel?) {
        this.data = data
        this.model = model
    }

    fun getDiscount(regular: Double, special: Double): Int {
        return ((regular - special) / regular * 100).toInt()
    }

    inner class ClickHandlers {
        fun moveToWishList(view: View, item: CartListItem) {
            model!!.moveToWishList(item)
            data!!.removeAt(item.position)
            notifyItemRemoved(item.position)
            notifyItemRangeChanged(item.position, data!!.size)
        }

        fun removeFromCart(view: View, item: CartListItem) {
            model!!.removeFromCart(item)
            data!!.removeAt(item.position)
            notifyItemRemoved(item.position)
            notifyItemRangeChanged(item.position, data!!.size)
        }

        fun increase(view: View, item: CartListItem) {
            item.qty = (Integer.parseInt(item.qty!!) + 1).toString()
            model!!.updateCart(item)
        }

        fun decrease(view: View, item: CartListItem) {
            if (Integer.parseInt(item.qty!!) == 1) {
                model!!.removeFromCart(item)
                data!!.removeAt(item.position)
                notifyItemRemoved(item.position)
                notifyItemRangeChanged(item.position, data!!.size)
            } else {
                item.qty = (Integer.parseInt(item.qty!!) - 1).toString()
                model!!.updateCart(item)
            }
        }
    }
}
